default/scripting/game_rules.focs.py

```diff
@@ -0,0 +1,134 @@
+GameRule(
```

Comment:
I just reproduce FOCS with minimal changes. I had idea to infer type from default value to remove `type` argument. If argument have no sense we can just not use it.

Reply:
> If argument have no sense we can just not use it.

If it is possible to add an argument that has no sense it will be added. It always works in that way. And of cause, it will confuse the user, because of the mismatch of the expectations and reality. We can protect ourselves with documentation and strict validation.  I'm not aware if documentation is in plans. But validation should not be very hard to implement.

Reply:
I've setup pep8 check. Not sure what could be done for stricter validation.

Reply:
I mean about the contnet.  Something like that.

Each type expects its own set of arguments. Some copy-paste issues and misunderstandings will get some problems.  I'd like to catch them on the CI stage.

```python
GameRule(
    name="XXX",
    description="XXX",
    category="BALANCE",
    type=str,
    max="x")
    
GameRule(
    name="RULE_TEST_STRING",
    description="RULE_TEST_STRING_DESC",
    category="TEST",
    type=bool,
    allowed=[true, false, "maybe"])
```

Reply:
We run unit tests for parsing code now so could check it here.